### PR TITLE
Redirecting on assurance level and displaying new text for bcsc

### PR DESCRIPTION
**Pull request by adodge-bcg**
# Description

Added new state to track where a transition page redirect came from, default to bcsc for use as failure redirect from BCSC failed or cancelled login. Added redirect from user confirmation if the assurance level is less than 3. BCSC currently does not allow assurance level less than 3 to redirect to application so may be redundant.

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Tested locally with storybook.
transitionBCSC

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## For bcgov contributors:

this PR fixes jira ticket: SPDBCSC-376 and SPDBCSC-297

### CODE REVIEW

File: src/ecrc-frontend/src/components/page/transition/Transition.stories.js [hunk 1]
```diff
@@ -11,9 +11,17 @@ const header = {
...
 };
 
-export const Default = () => <Transition header={header} />;
+const page = {
+  header
+};
+
+export const Default = () => <Transition page={page} />;
+
+export const NotWhitelisted = () => (
```
Comment (sdevalapurkar-bcgov) [inline src/ecrc-frontend/src/components/page/transition/Transition.stories.js]: should we add a story for bcsc case here?

File: src/ecrc-frontend/src/components/page/transition/Transition.test.js [hunk 2]
```diff
@@ -4,12 +4,22 @@ import { create } from "react-test-renderer";
...
+
+  test("Matches the default snapshot", () => {
+    const defaultTransition = create(<Transition page={page} />);
+    expect(defaultTransition.toJSON()).toMatchSnapshot();
+  });
+  test("Matches the non-whitelisted snapshot", () => {
+    const nonWhitelistTransition = create(
+      <Transition page={{ ...page, transitionReason: "notwhitelisted" }} />
+    );
+    expect(nonWhitelistTransition.toJSON()).toMatchSnapshot();
```
Comment (sdevalapurkar-bcgov) [inline src/ecrc-frontend/src/components/page/transition/Transition.test.js]: same here, a test case for bcsc

File: src/ecrc-frontend/src/components/page/transition/Transition.stories.js (hunk 1 above)
Comment (adodge-bcg) [reply src/ecrc-frontend/src/components/page/transition/Transition.stories.js]: BCSC is currently the default case so this page can be used as a redirect point on failure/cancel login at bcsc. Can rename Default to indicate this.

File: src/ecrc-frontend/src/components/page/transition/Transition.test.js (hunk 2 above)
Comment (adodge-bcg) [reply src/ecrc-frontend/src/components/page/transition/Transition.test.js]: Same here can change the name to reflect that default will display bcsc

File: src/ecrc-frontend/src/components/page/transition/Transition.stories.js (hunk 1 above)
Comment (sdevalapurkar-bcgov) [reply src/ecrc-frontend/src/components/page/transition/Transition.stories.js]: yes pls, thanks :)

File: src/ecrc-frontend/src/components/page/transition/Transition.test.js (hunk 2 above)
Comment (sdevalapurkar-bcgov) [reply src/ecrc-frontend/src/components/page/transition/Transition.test.js]: yup! would be great

**Pull request merged**